gluten-core/src/main/scala/io/glutenproject/execution/ScanTransformerFactory.scala

```diff
@@ -0,0 +1,135 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.glutenproject.execution
+
+import io.glutenproject.execution.FilterHandler.{flattenCondition, getLeftFilters}
+import io.glutenproject.expression.ExpressionConverter
+import io.glutenproject.sql.shims.SparkShimLoader
+
+import org.apache.spark.sql.connector.read.Scan
+import org.apache.spark.sql.execution.{FileSourceScanExec, FilterExec}
+import org.apache.spark.sql.execution.datasources.v2.{BatchScanExec, FileScan}
+
+import java.util.ServiceLoader
+import java.util.concurrent.ConcurrentHashMap
+
+import scala.collection.JavaConverters._
+
+object ScanTransformerFactory {
+
+  private val dataSourceV2TransformerMap = new ConcurrentHashMap[String, Class[_]]()
+
+  def createFileSourceScanTransformer(
```

Comment:
can we combine the two `createFileSourceScanTransformer` methods ?

Reply:
Of course, initially they were combined together, but we need to add some optional "Option" parameters.